src/mmw/apps/modeling/mapshed/calcs.py

```diff
@@ -204,6 +205,43 @@ def manure_spread(aeu):
     return n_list, p_list
 
 
+def ls_factors(lu_strms, areas, avg_slope):
+    results = [0.0] * len(lu_strms)
+    for i in range(len(lu_strms)):
+        if 0 <= avg_slope <= 1.0:
+            m = 0.2
+        elif 1.0 < avg_slope <= 3.5:
+            m = 0.3
+        elif 3.5 < avg_slope <= 4.5:
+            m = 0.4
+        else:
+            m = 0.5
+
+        results[i] = ls_factor(lu_strms[i] * KM_PER_M, areas[i], avg_slope, m)
```

Comment:
The contents of `lu_strms` here is not the actual length, but only a percentage. To get length in meters you'll need to multiply with `z.StreamLength`, and then convert to kilometers